File: Data Structure/Linked Lists/Linkedlist.cpp
#include<iostream>
using namespace std;

struct Node{
    int value;
    Node* next;
};

class Linkedlist{
    Node* head;
};


public:
    Linkedlist():head(){}

    void insertAtStart(int value)
    {
        Node* newNode = new Node();
        newNode->value = value;
        newNode->next = head;
        head = newNode;
    }

    void insertAtEnd(int value)
    {
        Node* newNode = new Node();
        newNode->value = value;
        newNode->next = NULL;
    }

    void deleteFromStart()
    {
        Node* temp = head;
        head = head->next;
        delete temp;
    }

    void deleteFromEnd()
    {
        if(!head->next)
        {
            delete head;
            head = NULL;
            return
        }
    }



int main()
{
    Linkedlist list;

    list.insertAtStart(1);

    list.insertAtEnd(10);

    list.deleteFromStart();
    list.display();

    list.deleteFromEnd();
    list.display();

    return 0;
}

File: Data Structure/Linked Lists/Linkedlist.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

    class Linkedlist:
        def __init__(self):
            self.head = None
    

        def insertAtStart(self, value):
            new_node = Node(value)
            new_node.next = self.hesd
            self.head = new_node

        def removeAtEnd(self):
            self.head = self.head.next

l = Linkedlist()
l.insertAtStart(1)
l.removeAtEnd()